src/helpers.js

```diff
@@ -83,7 +103,7 @@ function validatePoint(textBuffer, line, col) {
   }
 }
 
-export async function getDebugInfo(worker) {
+export async function getDebugInfo() {
```

Comment:
I like this a lot. Gets a ton of the worker logic and state management of the worker out the the `main` object, greatly simplifying things. Definitely going the right direction IMHO.